packages/site/src/components/PaymentChannelDetails.tsx

```diff
@@ -1,7 +1,206 @@
-type Props = {
-  paymentChannel: string;
-};
+import {
+  Box,
+  LinearProgress,
+  Typography,
+  Stack,
+  Button,
+  SvgIcon,
+} from "@mui/material";
+import { FC, useEffect, useMemo, useState } from "react";
+import { makeStyles } from "tss-react/mui";
+import { PhoneArrowUpRightIcon, UserIcon } from "@heroicons/react/24/outline";
+import { ChannelStatus } from "@statechannels/nitro-rpc-client/src/types";
 
-export default function PaymentChannelDetails({ paymentChannel }: Props) {
-  return <p>{paymentChannel}</p>;
+interface PaymentChannelDetails {
+  channelID: string;
+  payee: string;
+  payer: string;
+  paidSoFar: bigint | string;
+  remainingFunds: bigint | string;
+  status: ChannelStatus;
 }
+
+const useStyles = makeStyles()(() => ({
+  typography: {
+    marginTop: "0 !important",
+  },
+  leftPeer: {},
+  rightPeer: {
+    marginTop: "-6rem !important",
+  },
+  icons: {
+    lineHeight: "0 !important",
+    fontSize: "3rem",
+  },
+  iconLeft: {
+    fontSize: "6rem",
+  },
+  iconRight: {
+    fontSize: "2.5rem",
+  },
+}));
+
+const shortString = (value: string, count: number) => {
+  return `${value.slice(0, count)}...`;
+};
+
+const PaymentChannelDetails: FC<PaymentChannelDetails> = ({
+  channelID,
+  payee,
+  payer,
+  paidSoFar,
+  remainingFunds,
+  status,
+}: PaymentChannelDetails) => {
+  const [progress, setProgress] = useState<number>(0);
+  const { classes, cx } = useStyles();
+
+  const paidSoFarValue: bigint = useMemo(() => {
+    if (typeof paidSoFar === "string") {
+      return BigInt(parseInt(paidSoFar, 16));
+    }
+
+    return paidSoFar;
+  }, [paidSoFar]);
+
+  const remainingFundsValue: bigint = useMemo(() => {
+    if (typeof remainingFunds === "string") {
+      return BigInt(parseInt(remainingFunds, 16));
+    }
+
+    return remainingFunds;
+  }, [remainingFunds]);
+
+  useEffect(() => {
+    setProgress(
+      (Number(paidSoFarValue) /
+        (Number(remainingFundsValue) + Number(paidSoFarValue))) *
+        100
+    );
+  }, [paidSoFarValue, remainingFundsValue]);
+
+  const capitalizedStatus = useMemo(() => {
```

Comment:
Is this a premature optimization?

Reply:
For this - we can create a helper that takes some string as the argument and returns this string with capitalized first letter. This is not mandatory at all to have this function in the component

From the other side, we can keep it as it is

Reply:
I feel like this is overkill. I don't think modern browsers need `useMemo` for a string conversion function.